### ISSUE TITLE
Reactive variable placeholders appearing in content prior to rendering in an iframe.

### ISSUE THREAD

**Issue by bob2517** [Active-CSS/active-css#87]
When rendering Active CSS inline to an iframe from an $HTML ajax variable, at the point of render there are reactive variable placeholders and there really shouldn't be. If it was being rendered anywhere else then they would be expected to be there at that point, so the $HTML variable is acting correctly. It's like the core thinks it's doing the right thing, but then it's "Ohp! you're putting this in an iframe, it's all going to go Pete Tong. Doh."

Either something needs to not be happening - like don't put in placeholders if it's going to be in an iframe, if there's a way of working that out in advance, or this use case needs a different sort of variable for rendering in iframes. I'm leaning towards the different sort of variable, and it's leaning back at me and things are now getting weird.

I guess the placeholders could be converted back to variable names at the point of rendering to an iframe, but that's a bit hacky so don't do that.

**Comment by bob2517** [comment]
Leaning towards a new $HTML_NOVARS variable type. We don't want HTML to be escaped when rendered into an iframe so $HTML_ESCAPED isn't any use, but we definitely don't want reactive variables to be set up before they've even reached the document.

I don't want to slow up the core by introducing complex lookahead, lookbehind, looktotheside checks to see if iframe rendering is imminent, even if mixed in with other output about to happen, and ideally it should even be going anywhere near variable substitution code, so I think the new variable type is the way to go, as that can all be avoided and just be substituted at last minute render point.

I hereby declare that then there was $HTML_NOVARS...

**Comment by bob2517** [comment]
The only issue is variable assignment. I ran into this before when I set up the $HTML and $HTML_ESCAPED variables.

Hmmm... assigning escaped HTML to a variable is fine - it's escaped. Rendering regular HTML to a variable is fine, it isn't escaped and it's ok.

But rendering a string containing NOVARS. Hmmm... I don't think the last minute rendering is going to work.

The safest bet for a NOVARS declaration is just to do what you'd do if you wanted to output curly brackets as curly brakcets on a render and just put a backslash before them. Then it will work on variable assignment.

In theory it should need a removal on rendering to an iframe and that should be that.

But what about rendering intentionally escaped curly brackets to an iframe?

Hmmm.... ok we'll use an internal escaping method and just make sure these are removed at render point - regular render and iframe render.

The only problem with doing things like that though which I don't like is that it makes it harder to implement any sort of future user-based string manipulation with a NOVARS variable. Will ponder it a bit more. Thinking about it, that may not be a big deal - it's a bit niche and a rule could just be set to say that it acts intentionally like that. Once you start re-assigning variable types things can go weird anyway. HTML_NOVARS would be a specific variable type in Active CSS.
